lib/metadata/chordMetadata.hpp

```diff
@@ -392,6 +399,128 @@ class chordMetadata : public metadataObject {
         this->onehot_index[dim] = i;
         strncpy(this->onehot_name[dim], name.c_str(), CHORD_META_MAX_DIMNAME);
     }
+
```

Comment:
General comment on this class.   We now seem to have a slightly odd mix of old struct based logic for some items, and new json backed methods.  This is going to have some strange effects with serialization/deserialization.   As much as it's a huge pain, my feeling is we need to either commit entirely to changing this all over, or at the very least create serialization/deserialization which addresses _both_ the old components and the new components. 

Also in terms of naming.  I think we should rename this `frameMetadata`, since it's basically just taking on all possible types of frame metadata in the system.   We might someday want to breakup the classes just for clarity, but that's not obviously an improvement.

Reply:
yup, for anything to go forward this needs to all move into json. This pull request (and the work for it) was mostly to see if there are already objections in how the getters / setters work. I will move the leftover bits to json.